inference-engine/src/transformations/src/transformations/convert_precision.cpp

```diff
@@ -300,17 +300,22 @@ std::shared_ptr<Node> change_constant_precision(std::shared_ptr<opset4::Constant
     using src_type = typename element_type_traits<PREC_FROM>::value_type;
     using dst_type = typename element_type_traits<PREC_TO>::value_type;
 
-    std::vector<src_type> data(std::move(constant->get_vector<src_type>()));
+    const auto * src_data = constant->get_data_ptr<src_type>();
+    const auto size = shape_size(constant->get_shape());
+
+    auto new_constant = std::make_shared<ngraph::opset4::Constant>(PREC_TO, constant->get_shape());
+    auto * dst_data = const_cast<dst_type *>(reinterpret_cast<const dst_type *>(new_constant->get_data_ptr()));
+
     std::vector<dst_type> final_data;
-    std::transform(data.begin(), data.end(), std::back_inserter(final_data),
-                   [](src_type val) {
-                       if (val > std::numeric_limits<dst_type>::max()) {
-                           return std::numeric_limits<dst_type>::max();
-                       } else {
-                           return static_cast<dst_type>(val);
-                       }
-                   });
-    return std::make_shared<ngraph::opset4::Constant>(PREC_TO, constant->get_shape(), final_data);
+    for (size_t i = 0; i < size; ++i) {
+        const auto & val = src_data[i];
+        if (val > std::numeric_limits<dst_type>::max()) {
```

Comment:
I guess we should also add: `if (val < std::numeric_limits<dst_type>::min()) {`

Reply:
If val for example has uint64 type and dst_type int32 then comparison will be wrong as for comparison will be used type with bigger size -> uint64. So then min int32 value-2^32 will be converted to unsigned and comparison will be incorrect.

Reply:
But what if we want to convert int64 to int32?